### DISCUSSION TITLE
Question RE UE4/5 rendering

### DISCUSSION THREAD

**Discussion by Mfarshore** [italomandara/CXPatcher#197] [General]
What causes the floating polygons to appear, as in the screen shot attached. The image is from a game Astrometica recently released. Running under CrossOver24 with UE4 via cxpatcher. Tried it patching Base and Latest MVK+UE4 hack and without Patch DXVK with no luck. This is something I have seen in both UE4 and UE5. Any idea how to get around it using UE scripting additions?

![Screenshot 2024-05-26 at 4 10 57 PM](https://github.com/italomandara/CXPatcher/assets/134976176/36e9e690-af4d-444c-a7bd-2e3f3923fc6d)

**Comment by Mfarshore** [comment]
I found that enabling MSync will fix this problem, for this particular game. The graphics pipeline was getting misaligned with the CPU.

**Comment by italomandara** [comment]
DXVK is using MoltenVK as a vulkan translation layer, MVK isn't perfect as we speak it has probably to do with synchronization, sice MSYNC fixes it (which i almost always recommend) I'm not sure why MSYNC affects the GPU, but it's definitely an issue in MoltenVK.

**Comment by Mfarshore** [comment]
Yeah I realized that it involved the shader pipeline and MoltenVK. When I saw that the pipeline was  getting backup as the problems started I just took a stab and activated MSync on a hunch. MSync doesn’t affect the GPU directly. It just permits the shader pipeline to become unclogged so that the GPU can handle tasks in a timely manner.  I watched the various HUDs before and after starting MSync and the GPU usage definitely went down once the pipeline was unclogged.

On another matter CX 25 is out today I think. About how long until CXPatcher catches up?

**Comment by italomandara** [comment]
I'll start working on it tomorrow
